tesseract-clickhouse/src/lib.rs

```diff
@@ -28,7 +30,27 @@ impl Clickhouse {
 
         let options = options
             // Ping timeout is necessary, because under heavy load (100 requests
-            // simultenously, each one taking 5s ordinarily) the client will timeout.
+            // simultaneously, each one taking 5s ordinarily) the client will timeout.
+            .ping_timeout(Duration::from_millis(PING_TIMEOUT));
+
+        let pool = Pool::new(options);
+
+        Ok(Clickhouse {
+            pool,
+        })
+    }
+
+    pub fn from_url(url: &str) -> Result<Self, Error> {
+        let options = match Options::from_str(
```

Comment:
I think that you can use `parse` and `?` here.

`Options::from_str` is usually written as `let x: Options = "".parse()`, or with turbofish `let x = "".parse::<Options>()`.

the match statement is basically the old `try!` which is now `?`. failure's `Error` is a type which most other errors can be converted into (with early return) using `?`.

So the end product might be something like `let options = format!("tcp://{}", url).parse::<Options>()?`